Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2151,22 +2152,22 @@ - (NSAttributedString *)collectionView:(JSQMessagesCollectionView *)collectionVi
             return [[NSAttributedString alloc]
                 initWithString:NSLocalizedString(@"MESSAGE_STATUS_FAILED", @"message footer for failed messages")];
         } else if (outgoingMessage.messageState == TSOutgoingMessageStateSentToService) {
-            NSString *text = (outgoingMessage.wasDelivered
-                    ? NSLocalizedString(@"MESSAGE_STATUS_DELIVERED", @"message footer for delivered messages")
-                    : NSLocalizedString(@"MESSAGE_STATUS_SENT", @"message footer for sent messages"));
-            NSAttributedString *result = [[NSAttributedString alloc] initWithString:text];
-            if ([OWSReadReceiptManager.sharedManager areReadReceiptsEnabled] && outgoingMessage.wasDelivered
-                && outgoingMessage.recipientReadMap.count > 0) {
-                NSAttributedString *checkmark = [[NSAttributedString alloc]
-                    initWithString:@"\uf00c "
-                        attributes:@{
-                            NSFontAttributeName : [UIFont ows_fontAwesomeFont:10.f],
-                            NSForegroundColorAttributeName : [UIColor ows_materialBlueColor],
-                        }];
-                NSAttributedString *spacing = [[NSAttributedString alloc] initWithString:@" "];
-                result = [[checkmark rtlSafeAppend:spacing referenceView:self.view] rtlSafeAppend:result
-                                                                                    referenceView:self.view];
+            NSString *text;
+            if (outgoingMessage.wasDelivered) {
+                NSNumber *_Nullable firstRecipientReadTimestamp = [outgoingMessage firstRecipientReadTimestamp];
+                if ([OWSReadReceiptManager.sharedManager areReadReceiptsEnabled] && firstRecipientReadTimestamp) {
+                    text = NSLocalizedString(@"MESSAGE_STATUS_READ", @"message footer for read messages");
+                    text = [text rtlSafeAppend:@" " referenceView:self.view];
```

Comment:
I'm leaning toward using `rtlSafeAppend()` everywhere we can, e.g. situations where we don't anticipate the localization to affect ordering beyond LTR-v.-RTL.  This should significantly simplify the job for our translators.